src/BatchRequestContent.ts

```diff
@@ -0,0 +1,460 @@
+                        return false;
+                    }
+                }
+                cur = iterator.next();
+            }
+            return true;
+        };
+        if (requests.size === 0) {
+            let error = new Error("Empty requests map, Please provide at least one request.");
+            error.name = "Empty Requests Error";
+            throw error;
+        }
+        return (isParallel(requests) || isSerial(requests) || isSame(requests));
+    }
+
+    /**
+     * @private
+     * @static
+     * @async
+     * Converts Request Object instance to a JSON
+     * @param {IsomorphicRequest} request - The IsomorphicRequest Object instance
+     * @return A promise that resolves to JSON representation of a request
+     */
+    private static async getRequestData(request: IsomorphicRequest): Promise<RequestData> {
+        let requestData = <RequestData>{};
+        let hasHttpRegex = new RegExp("^https?:\/\/");
+        if (hasHttpRegex.test(request.url)) {
+            // Stripping off hostname, port and url scheme
+            requestData.url = "/" + request.url.split(/.*?\/\/.*?\//)[1];
+        } else {
+            requestData.url = request.url;
+        }
+        requestData.method = request.method;
+        let headers = {};
+        request.headers.forEach((value, key) => {
+            headers[key] = value;
+        });
+        if (Object.keys(headers).length) {
+            requestData.headers = headers;
+        }
+        if (request.method === RequestMethod.PATCH || request.method === RequestMethod.POST || request.method === RequestMethod.PUT) {
+            requestData.body = await BatchRequestContent.getRequestBody(request);
+        }
+        /**
+         * Check any other property needs to be used from the Request object and add them
+         */
+        return requestData;
+    }
+
+    /**
+     * @private
+     * @static
+     * @async
+     * Gets the body of a Request object instance
+     * @param {IsomorphicRequest} request - The IsomorphicRequest object instance
+     * @return The Promise that resolves to a body value of a Request
+     */
+    private static async getRequestBody(request: IsomorphicRequest): Promise<any> {
+        let bodyParsed: boolean = false,
+            body;
+        try {
+            let cloneReq = request.clone();
+            body = await cloneReq.json();
+            bodyParsed = true;
+        } catch (e) {
+
+        }
+        if (!bodyParsed) {
+            try {
+                if (typeof Blob !== "undefined") {
+                    let blob = await request.blob();
+                    let reader = new FileReader();
+                    body = await new Promise(resolve => {
+                        reader.addEventListener("load", function () {
+                            let dataURL = <string>reader.result,
+                                regex = new RegExp("^\s*data:(.+?\/.+?(;.+?\=.+?)*)?(;base64)?,(.*)\s*$"),
```

Comment:
It is generally a good idea to add a couple of examples in the comments for regular expression patterns.